src/table/utils/styling-utils.js

```diff
@@ -25,58 +27,78 @@ export const SELECTION_STYLING = {
 
-export const getBaseStyling = (styleObj, theme) => ({
-  color: getColor(styleObj.fontColor, STYLING_DEFAULTS.FONT_COLOR, theme),
-  fontSize: styleObj.fontSize || STYLING_DEFAULTS.FONT_SIZE,
-  padding: styleObj.fontSize ? `${styleObj.fontSize / 2}px ${styleObj.fontSize}px` : STYLING_DEFAULTS.PADDING,
-});
+export const getBaseStyling = (styleObj, objetName, theme) => {
+  const backgroundColor = theme.getStyle('object', 'straightTable', 'backgroundColor');
+  const fontFamily = theme.getStyle('object', `straightTable.${objetName}`, 'fontFamily');
+  const color = theme.getStyle('object', `straightTable.${objetName}`, 'color');
+  const fontSize = theme.getStyle('object', `straightTable.${objetName}`, 'fontSize');
+
+  const baseStyle = {
+    backgroundColor,
+    fontFamily,
+    color: styleObj?.fontColor ? getColor(STYLING_DEFAULTS.FONT_COLOR, theme, styleObj?.fontColor) : color,
+    fontSize: styleObj?.fontSize || fontSize,
+    padding: styleObj?.fontSize && `${styleObj?.fontSize / 2}px ${styleObj?.fontSize}px`,
+  };
+  // Remove all Undefined Values from an Object
+  Object.keys(baseStyle).forEach((key) => baseStyle[key] == null && delete baseStyle[key]);
+  return baseStyle;
+};
 
 // Both index === -1 and color === null must be true for the property to be unset
 export const isUnset = (prop) => !prop || JSON.stringify(prop) === JSON.stringify({ index: -1, color: null });
 
 export function getHeadStyle(layout, theme) {
   const header = layout.components?.[0]?.header;
-  return header ? getBaseStyling(header, theme) : { padding: STYLING_DEFAULTS.PADDING };
+
+  return getBaseStyling(header, 'header', theme);
 }
 
 export function getBodyStyle(layout, theme) {
   const content = layout.components?.[0]?.content;
-  if (!content) return { padding: STYLING_DEFAULTS.PADDING };
 
   // Cases when hoverEffect is true:
   // 1. There is no hover font color but a hover background color,
-  // when hovering, the hover font color becomes white when the hover background color is a dark color
-  // or the hover font color stays the same as whetever the font color is when the hover background color is a light color.
+  // when hovering, the hover font color becomes white when the hover
+  // background color is a dark color or the hover font color stays
+  // the same as whatever the font color is when the hover background
+  // color is a light color.
   // 2. There is a hover font color but no hover background color,
-  // when hovering, only a hover font color is applied and the hover background color disappears.
-  // 3. There is no hover font color and no hover background color, when hovering, the defalut hover effect (light gray backgournd)
-  // 4. There are both hover font and background colors, when hovering, the hover font and background colors take effect.
+  // when hovering, only a hover font color is applied and the hover
+  // background color disappears.
```

Comment:
maybe say that the hover background color is transparent. disappears sounds a bit magic

Reply:
in this case, the value of hoverBackgroundColor is ''. I would not say it is transparent. 
How about 'There is a hover font color but no hover background color, when hovering, only the font color is applied and background color is the same as before'

Reply:
same as what? the background color or the default hover background color? `There is a hover font color but no hover background color, when hovering, only the font color is applied and no hover background is shown` or similar

Reply:
that's better. Thanks. Updated.